core/resource/resource.go

```diff
@@ -20,6 +20,7 @@ type Repository interface {
 	Create(ctx context.Context, resource Resource) (Resource, error)
 	List(ctx context.Context, flt Filter) ([]Resource, error)
 	Update(ctx context.Context, id string, resource Resource) (Resource, error)
+	GetByNamespace(ctx context.Context, name string, ns namespace.Namespace) (Resource, error)
```

Comment:
isn't there a string identifier that can be used to reference a namespace here? 

passing namespace struct leaves lot of things ambiguous:

1. what part of the struct are used in the lookup? 
2. will the query use CreatedAt and UpdatedAt as well? (probably not, but it's still possible to be done by the repo implementation)

Reply:
we can get resource by combination between resource name and:
1. namespace_id
2. backend + type
3. backend

thus, using struct

Reply:
signed off during standup